Source/KSCrash/Recording/Tools/KSJSONCodec.c

```diff
@@ -1137,8 +1137,19 @@ int ksjsoncodec_i_decodeElement(const char** ptr,
                 return KSJSON_ERROR_INCOMPLETE;
             }
 
+            // our buffer is not necessarily NULL-terminated, so
+            // it would be undefined to call sscanf/sttod etc. directly.
+            // instead we create a temporary string.
             double value;
-            sscanf(start, "%lg", &value);
+            size_t len = (size_t)(*ptr - start);
+            char * buf = malloc(len + 1);
```

Comment:
Actually, this has the potential to allocate a buffer as big as the entire JSON string, doesn't it? Let's restrict the max size to something like 30 characters.